tests/merkle_proof_test.rs

```diff
@@ -80,6 +80,33 @@ pub mod root {
 
         Ok(())
     }
+
+    #[test]
+    pub fn should_return_error_not_enough_hashes_to_calculate_root() {
+        let test_data = common::setup();
+        let leaf_hashes = &test_data.leaf_hashes;
+        let expected_root = test_data.expected_root_hex.clone();
+        let indices_to_prove = vec![3, 4];
+
+        let leaves_to_prove: Vec<[u8; 32]> = indices_to_prove
+            .iter()
+            .map(|i| leaf_hashes.get(*i).unwrap().clone())
+            .collect();
+
+        let merkle_tree = MerkleTree::<Sha256>::from_leaves(&test_data.leaf_hashes);
+        let proof = merkle_tree.proof(&indices_to_prove);
+        let extracted_root = proof.root(
+            &indices_to_prove,
+            &leaves_to_prove,
+            test_data.leaf_values.len() + 1,
```

Comment:
@witter-deland, could you pls look into this test again.
the `+1` shouldn't be here iiuc. When I remove it, this test fails.

Reply:
+1 means incorrect proof hash count, why did you remove it?

Reply:
i see, sry i was confused for a second. I hope you don't mind if a change the test to actually remove the hash https://github.com/antouhou/rs-merkle/pull/31